suites/apis/googleServiceAccountRemovalTest.js

```diff
@@ -0,0 +1,471 @@
+  ).to.have.property('statusCode', 403);
+
+  chai.expect(unlockRes, 'Could not unlock project').to.be.true;
+}).retry(2);
+
+
+Scenario('SA removal job test: user does not have access to data @reqGoogle', async (fence, users, google, files) => {
+  // test invalid project because the user does not have access to the data
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  let registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
+
+  // Make the project invalid
+  console.log(`Running useryaml job with ${Commons.userAccessFiles.newUserAccessFile2}`);
+  Commons.setUserYaml(Commons.userAccessFiles.newUserAccessFile2);
+  bash.runJob('useryaml-job');
+  // now user0 does not have access to project QA anymore
+
+  // Run clean up job
+  let jobRes = checkAndCleanSA();
+
+  // Try to access data
+  let [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+  user0AccessQARes = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Clean up
+  console.log('cleaning up');
+
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  await fence.do.deleteGoogleServiceAccount(
+    users.user0,
+    googleProject.serviceAccountEmail,
+  );
+
+  await fence.do.unlinkGoogleAcct(
+    users.user0,
+  );
+
+  // Unlock the project
+  let unlockRes = await google.unlockGoogleProject(googleProject);
+
+  console.log(`Running usersync job`);
+  bash.runJob('usersync');
+
+  // Asserts
+  fence.ask.detected_invalid_google_project(jobRes, fence.props.monitorSAJobLog.noDataAccess);
+
+  chai.expect(user0AccessQARes,
+    'User should NOT have bucket access after clean up job'
+  ).to.have.property('statusCode', 403);
+
+  chai.expect(unlockRes, 'Could not unlock project').to.be.true;
+}).retry(2);
+
+
+Scenario('SA removal job test: SA has external access @reqGoogle', async (fence, users, google, files) => {
+  // test invalid project because the SA has an external key set up
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  let registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
```

Comment:
we may want to lock before registering the SA. Fence does a ton of checks against the project to make sure the SA can be registered, if it's already locked and invalid, registration may fail